src/components/PopUpWithConfirmation.js

```diff
@@ -0,0 +1,33 @@
+import PopUp from "./PopUp";
+
+export default class PopUpWithConfirmation extends PopUp {
+  constructor(popUpSelector) {
+    super({ popUpSelector });
+    this._popUpForm = this._popUpElement.querySelector(".modal__form");
+    this._submitButton = this._popUpForm.querySelector(".modal__button");
+    this._submitButtonText =
+      this._submitButton.querySelector("#delete-card-modal");
+  }
+
+  setSubmitAction(action) {
+    this._handleFormSubmit = action;
+  }
+
+  setLoading(submit, loadingText = "Saving...") {
```

Comment:
Great that you've implemented the logic for rendering loading via the class methods